include/aspect/utilities.h

```diff
@@ -392,4 +392,27 @@ namespace aspect
   }
 }
 
+namespace dealii
+{
+  namespace Utilities
+  {
+    namespace MPI
+    {
+      /**
+       * Perform an MPI sum over the quantities in the given Tensor<1,dim>
+      */
+      template<int dim>
+      Tensor<1,dim> 
+      reduce_tensor(const Tensor<1,dim> &local, const MPI_Comm &mpi_communicator);
+
+      /**
+       * Perform an MPI sum over the quantities in the given SymmetricTensor<2,dim>
+       */
+      template<int dim>
+      SymmetricTensor<2,dim> 
+      reduce_symmetric_tensor(const SymmetricTensor<2,dim> &local, const MPI_Comm &mpi_communicator);
```

Comment:
Yes, fair enough, I wasn't quite sure what you meant.  If these look
reasonable to you I can open a separate PR against deal.II.  Are they as
general as you would want?

If we did that, I would still have to implement a fallback here if the
deal.II version was not the dev version.

And yes, you are right, sum is a more descriptive name.

On Tue, Apr 21, 2015 at 6:51 PM, Wolfgang Bangerth <notifications@github.com
> wrote:

> In include/aspect/utilities.h
> <https://github.com/geodynamics/aspect/pull/305#discussion_r28838929>:
>
> > +  {
> > +    namespace MPI
> > +    {
> > +      /**
> > +       * Perform an MPI sum over the quantities in the given Tensor<1,dim>
> > +      */
> > +      template<int dim>
> > +      Tensor<1,dim>
> > +      reduce_tensor(const Tensor<1,dim> &local, const MPI_Comm &mpi_communicator);
> > +
> > +      /**
> > +       * Perform an MPI sum over the quantities in the given SymmetricTensor<2,dim>
> > +       */
> > +      template<int dim>
> > +      SymmetricTensor<2,dim>
> > +      reduce_symmetric_tensor(const SymmetricTensor<2,dim> &local, const MPI_Comm &mpi_communicator);
>
> I thought of these as going into deal.II's dealii::Utilities::MPI where
> the other sum functions already are. I think sum would actually be a
> descriptive name for the function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/aspect/pull/305/files#r28838929>.
>